src/weather_model_graphs/create/archetype.py

```diff
@@ -67,39 +73,49 @@ def create_graphcast_graph(xy_grid, refinement_factor=3, max_num_levels=None):
         xy=xy_grid,
         m2m_connectivity="flat_multiscale",
         m2m_connectivity_kwargs=dict(
-            refinement_factor=refinement_factor, max_num_levels=max_num_levels
+            grid_refinement_factor=grid_refinement_factor,
+            level_refinement_factor=level_refinement_factor,
+            max_num_levels=max_num_levels,
         ),
-        m2g_connectivity="nearest_neighbour",
-        g2m_connectivity="nearest_neighbours",
+        g2m_connectivity="within_radius",
+        m2g_connectivity="nearest_neighbours",
```

Comment:
I don't think I have ever read a thorough explanation, but mostly used this for historical reasons as well. One motivation could be based on what we think g2m and m2g does. g2m is supposed to aggregate grid information up to the mesh. It could then be useful to include a large area around each mesh node as the "information aggregation window", and overlaps in this are not a problem. For m2g, its purpose is to extract the information from the mesh to determine the final prediction in each grid node. At this point we expect this information to be localised to the closest mesh nodes, so in a sense m2g only performs a fancy interpolation between the closest mesh nodes. If we connect to the closest mesh nodes we don't expect mesh nodes further away to contribute with more infromation.